src/components/widgets/Toggle.tsx

```diff
@@ -0,0 +1,235 @@
+    const themeColor = theme && themeRole && theme.palette[themeRole];
+    const mainColor = themeColor?.hue ?? blue;
+    const mainLevel = themeColor?.level ?? 400;
+    const labelColor = gray[600];
+    const disabledColor = gray[300];
+
+    const defaultStyleSpec: ToggleStyleSpec = {
+      container: {},
+      default: {
+        off: {
+          backgroundColor: "none",
+          knobColor: mainColor[mainLevel],
+          borderColor: mainColor[mainLevel],
+          labelColor,
+        },
+        on: {
+          backgroundColor: mainColor[mainLevel],
+          knobColor: "white",
+          borderColor: mainColor[mainLevel],
+          labelColor,
+        },
+      },
+      hover: {
+        off: {
+          backgroundColor: mainColor[100],
+          knobColor: mainColor[mainLevel],
+          borderColor: mainColor[mainLevel],
+          labelColor,
+        },
+        on: {
+          backgroundColor: mainColor[Math.min(mainLevel + 200, 900)],
+          knobColor: "white",
+          borderColor: mainColor[Math.min(mainLevel + 200, 900)],
+          labelColor,
+        },
+      },
+      disabled: {
+        off: {
+          backgroundColor: "none",
+          knobColor: disabledColor,
+          borderColor: disabledColor,
+          labelColor,
+        },
+        on: {
+          backgroundColor: disabledColor,
+          knobColor: "white",
+          borderColor: disabledColor,
+          labelColor,
+        },
+      },
+    };
+
+    return merge({}, defaultStyleSpec, styleOverrides);
+  }, [styleOverrides, theme, themeRole]);
+
+  /**
+   * The CSS styles that should be applied can depend
+   * on (1) whether or not the component is
+   * focused/hovered, (2) which option is selected,
+   * and (3) whether the toggle is disabled.
+   */
+  const currentStyles = useMemo(() => {
+    const selectedOption = state ? "on" : "off";
+
+    return disabled
+      ? styleSpec.disabled[selectedOption]
+      : styleSpec[hoverState][selectedOption];
+  }, [hoverState, state, styleSpec, disabled]);
+
+  const ariaLabel = useMemo(() => {
+    if (label) return label + " Toggle";
+    else return "Toggle";
+  }, [label]);
+
+  const width = size === "medium" ? 40 : 20;
+  const height = width / 2;
+  const borderWidth = 2;
+  const knobSize = size === "medium" ? 12 : 5;
+  const knobOffset = size === "medium" ? 2 : 1;
+
+  return (
+    <div
+      css={{
+        display: "flex",
+        alignItems: "center",
+        pointerEvents: disabled ? "none" : "auto",
+        fontFamily: primaryFont,
+        fontSize: 13,
+        fontWeight: 400,
+        ...styleSpec.container,
+      }}
+    >
+      {label && labelPosition === "left" && (
+        <span
+          css={{
+            marginRight: size === "medium" ? 10 : 5,
+            color: currentStyles.labelColor,
+          }}
+        >
+          {label}
+        </span>
+      )}
+      <div
+        role="switch"
+        aria-label={ariaLabel}
```

Comment:
My understanding of `aria-label` is that, absent of an actual label, this implementation makes sense. But if we are rendering the span with a label, then we'd want to associate that label with the toggle via the `aria-labelledby` property. [This is a good, quick summary](https://accessibleweb.com/question-answer/when-to-use-aria-label-vs-aria-labelledby/) of the difference between the two.

@dmfalke may have a stronger opinion on whether or not it makes a big enough difference to warrant what I perceive to be the extra conditional logic to implement, but it's worth thinking about at least.

Reply:
Implemented!